[Jollybomber] iP

Jollybomber opened the pull request:
### iP PR Update

Nave, your personalized assistant, has received a significant update to improve your experience 🎉. Below is a summary of the changes and new features:

**Key Updates:**
- Enhanced user interface
- New functionalities added

**New Features Include:**
- **Custom Commands**: Navigate tasks with commands like `todo`, `deadlines`, and `events`.
- **Task Management**: Track and manage the completeness of your tasks.
- **Clashing events**: Checks if events are clashing before adding.

**Example Code:**
Here's the `main` method:
```java
public static void main(String[] args) {
       Nave bot = new Nave();
       bot.run();
}
```

To-Do List:

- [ ] Update user guide with new commands
- [ ] Collect user feedback on performance improvements

> “The future belongs to those who believe in the beauty of their dreams.” — [Eleanor Roosevelt] (https://www.brainyquote.com/quotes/eleanor_roosevelt_100940)

Feel free to reach out if you have any questions or need assistance!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Jollybomber committed:
Renamed chatbot to Nave and changed greeting message.

Jollybomber committed:
Renamed chatbot to Nave and changed greeting message.

Jollybomber committed:
Feat: Nave now echoes user input and exits when bye is typed

Jollybomber committed:
Feat: User can now add tasks and list tasks

Jollybomber committed:
Feat: User can now mark and unmark tasks as done. Refactored code to follow OOP principles better.

Jollybomber committed:
Feat: Added 3 different task types: todo, event, deadline and made task abstract. Added a /help function to list all available commands.

Jollybomber committed:
Feat: added I/O redirection testing

Jollybomber committed:
Feat: added error handling

Jollybomber committed:
Feat: added delete function

Jollybomber committed:
Edited exception class

Jollybomber committed:
Feat: added save function

Jollybomber committed:
Feat: added ability to add dates to deadlines, added ability to delete tasks that are saved locally

Jollybomber committed:
Merge pull request #1 from Jollybomber/branch-Level-8

Branch level 8

Jollybomber committed:
Feat: utilised OOP to abstract some functionalities in separate classes.

Jollybomber committed:
Feat: utilised OOP to abstract some functionalities in separate classes.

Jollybomber committed:
Feat: utilised OOP to abstract some functionalities in separate classes.

Jollybomber committed:
Feat: UI class

Jollybomber committed:
Merge pull request #2 from Jollybomber/more-oop

More oop

Jollybomber committed:
Feat: Nave package

Jollybomber committed:
Merge branch 'master' into add-gradle-support

Jollybomber committed:
Merge pull request #3 from Jollybomber/add-gradle-support

Add gradle support

Jollybomber committed:
Feat: Run Nave with Gradle

KennethTeo2002 commented on .gitignore:
```diff
@@ -15,3 +15,4 @@ bin/
 
 /text-ui-test/ACTUAL.TXT
 text-ui-test/EXPECTED-UNIX.TXT
```
Could standardize usage of "/" in front of file paths.

KennethTeo2002 commented on src/main/java/Nave/Deadline.java:
```diff
@@ -0,0 +1,60 @@
+        if (correctMatcher.matches()) {
+            LocalDate date = LocalDate.parse(correctMatcher.group(3));
+            return new Deadline(correctMatcher.group(1), date);
+        } else if (wrongMatcher1.matches()) {
```
Nice regex checking! Might be need to take note that it is hard to maintain as requirements changes.

riccoljy replied:
Agreed; seems extremely detailed and fail-safe! Agree with the part on it possibly being be hard to upkeep too.

KennethTeo2002 commented on src/main/java/Nave/Deadline.java:
```diff
@@ -0,0 +1,60 @@
+        }
+    }
+
+    public String creationResponse() {
```
method name should be an action like ```returnCreationResponse```. Same for other files.

KennethTeo2002 commented on src/main/java/Nave/Deadline.java:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    public String creationResponse() {
+        return "Ok! I've added a new task with a deadline:\n" + this +
```
"+" sign should be indented to the next line. Same for other files.

KennethTeo2002 commented on src/main/java/Nave/Nave.java:
```diff
@@ -0,0 +1,74 @@
+import java.util.Scanner;
+
+public class Nave {
+    private final TaskList tasks;
```
```taskList``` is a single object, thus it should not be named as a plural.

KennethTeo2002 commented on src/main/java/Nave/Task.java:
```diff
@@ -0,0 +1,41 @@
+package Nave;
+
+abstract public class Task {
```
```public``` should come before ```abstract```

KennethTeo2002 commented on src/main/java/Nave/TaskList.java:
```diff
@@ -0,0 +1,68 @@
+import java.util.ArrayList;
+
+public class TaskList {
+    private final ArrayList<Task> list;
```
Typically we try to avoid using the variable name ```list``` as it could cause confusion with ```java.util.List```

KennethTeo2002 commented on src/main/java/Nave/Todo.java:
```diff
@@ -0,0 +1,36 @@
+import java.util.regex.Pattern;
+
+public class Todo extends Task {
+    public Todo (String name) {
```
Should not have spacing between method name and ().

KennethTeo2002 commented on src/main/java/Nave/Ui.java:
```diff
@@ -0,0 +1,37 @@
+        System.out.println(formatResponse("Goodbye :( Come visit me again"));
+    }
+
+    public void helpMessage() {
```
help and unsure messages should have an action.

KennethTeo2002 reviewed (commented):
LGTM.
Minor naming convention changes needed, other than that, good work.

Jollybomber committed:
Feat: JUnit testing with Gradle

Jollybomber committed:
Feat: JAR file creation

Jollybomber committed:
Feat: Added JavaDocs

Jollybomber committed:
Feat: Followed coding standard

Jollybomber committed:
Feat: Added feature to find task based on keyword

Jollybomber committed:
Merge pull request #4 from Jollybomber/branch-A-JavaDoc

Feat: Added JavaDocs

Jollybomber committed:
Merge branch 'master' into branch-A-CodingStandard

Jollybomber committed:
Merge pull request #5 from Jollybomber/branch-A-CodingStandard

Feat: Followed coding standard

Jollybomber committed:
Merge branch 'master' into branch-Level-9

Jollybomber committed:
Merge pull request #6 from Jollybomber/branch-Level-9

Feat: Added feature to find task based on keyword

riccoljy commented on data/tasks.txt:
Perhaps this need not be tracked/staged/pushed? This could be unnecessary for if someone clones/forks this repo?

riccoljy commented on src/main/java/Nave/Deadline.java:
```diff
@@ -0,0 +1,112 @@
+            throw new WrongInputException("You forgot to specify a due date!");
+        }  else {
+            //Shouldn't reach if all error cases handled
+            throw new WrongInputException("Something's wrong!");
```
I like how detailed the exceptions are and how there's a 'shouldn't be reached' clause but still considered and implemented!

riccoljy commented on src/main/java/Nave/Nave.java:
```diff
@@ -0,0 +1,103 @@
+                    String keyword = parser.parseFind(userInput);
+                    ui.showResponse(tasks.findTasks(keyword));
+                    break;
+            }
```
I like how all the cases can be easily read! I believe that Checkstyle/the Java Style Guide indentation of "case" is in line with "switch", though.

riccoljy commented on src/main/java/Nave/Parser.java:
Awesome Regex matching!

riccoljy commented on src/main/java/Nave/Task.java:
```diff
@@ -0,0 +1,102 @@
+ */
+abstract public class Task {
+    private final String name;
+    private boolean done;
```
Minor nit but I believe `is_` should be the standard for boolean (like `isDone`), but `done` seems unambiguous enough.

riccoljy commented on src/main/java/Nave/Task.java:
```diff
@@ -0,0 +1,102 @@
+                ? "[X] "
+                : "[] ";
+        return checked + name;
+    }
```
Nice indentation for ternary! Made it easy to read.

riccoljy commented on src/main/java/Nave/TaskList.java:
```diff
@@ -0,0 +1,129 @@
+     * @return a string message stating the number of tasks
+     */
+    public String countTasks() {
+        return "You have " + tasks.size() + " tasks now.";
```
Maybe can account for singular task?

riccoljy commented on src/main/java/Nave/TaskList.java:
```diff
@@ -0,0 +1,129 @@
+     *
+     * @return a string containing all tasks in the list
+     */
+    public String listItems() {
```
`listItems` seems like a method name that would `System.out.println` something, which is the job of `Ui.java`, but I realised that this does not print anything only after reading the entire method. Perhaps a better method name, like `listItemsToString`?

riccoljy commented on src/main/java/Nave/TaskList.java:
```diff
@@ -0,0 +1,129 @@
+                : "This item was already marked as done previously!";
+        } catch (IndexOutOfBoundsException e) {
+            return "This task does not exist!";
+        }
```
I like the ternary operator and the out of bounds exception!

riccoljy commented on src/main/java/Nave/TaskStorage.java:
```diff
@@ -0,0 +1,127 @@
+                System.exit(0);
+            }
+        }
+    }
```
Might want to have a catch/default branch for if the file is found but corrupted (ie: split.length is not 1, 2, or 3)

riccoljy reviewed (commented):
Just minor nits like formatting and possibly bugs but other than that, LGTM!

Jollybomber committed:
Fix: CheckStyle errors

Jollybomber committed:
Merge pull request #7 from Jollybomber/branch-A-CheckStyle

Fix: CheckStyle errors

Jollybomber committed:
Feat: Added JavaFX UI for Nave

Jollybomber committed:
Merge pull request #8 from Jollybomber/branch-Level-10

Feat: Added JavaFX UI for Nave

Jollybomber committed:
Feat: Added assertions to ensure that index submitted by user is in the right range.

Jollybomber committed:
Refactored code to follow SLAP.

There was code within Nave class that handled the core functionality of the bot, calling the parser, ui and task storage.

The code that handled getting the user input and parsing it, subsequently outputting a response has been abstracted out.

The code for the function run(), now better follows the SLAP principle.

In addition, duplicated code for parsing the user input for the GUI vs the command line has been consolidated to prevent code duplication.

Jollybomber committed:
Merge branch 'master' into branch-A-CodeQuality

Jollybomber committed:
Merge pull request #9 from Jollybomber/branch-A-CodeQuality

Branch a code quality

Jollybomber committed:
Merge branch 'master' into branch-A-CodeQuality

Jollybomber committed:
Added feature to check if event to be added clashes with other events in the list.

Changed event to accept dates like Deadline and store them as LocalDate to assist in adding feature stated above.

Nave now lists clashing events if the dates overlap and will not add the new event if there is a clash.

Jollybomber committed:
Merge pull request #10 from Jollybomber/branch-BCD-Extension

Added feature to check if event to be added clashes with other events…

Jollybomber committed:
Merge pull request #12 from Jollybomber/branch-A-CodeQuality

Merge branch 'master' into branch-A-CodeQuality

Jollybomber committed:
Add title to javafx window

Jollybomber committed:
Add ui screenshot

Jollybomber committed:
Update readme

Jollybomber committed:
Update readme

Jollybomber committed:
Update readme

Jollybomber committed:
Add ability for data file to be auto created

Jollybomber committed:
Update file creation
